VatSpace/main.lua

```diff
@@ -36,7 +36,14 @@ end
 
 local function restartGameForNextLevel()
     if gameTimer then
-        stopTimer()        
+        stopTimer() 
+        asteroids.destroy()
+        asteroids.create()
+        cats.cat:toBack()  
+        ui.goalTextGroup:toFront()
+        ui.morawieckiGroup:toFront()
+        ui.progressViewGroup:toFront()
```

Comment:
Cool methods, but as I see in documentation, its moving objects only in their groups. Are you sure that is working for our problem?

Reply:
I read documentation :) And it's work fine in our case (except for cat but this is separate topic).

Reply:
What do you think about encapsulate this 3 functions to new function? I think i'ts even more needed then recreate() function you mention before.

Reply:
Definitely, it should't be in main.

Reply:
Done.